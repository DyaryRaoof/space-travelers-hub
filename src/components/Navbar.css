.header-title {
  font-size: 30px;
}

.planet-logo {
  height: 50px;
  padding-right: 10px;
}

.nav-link {
  color: #0679f9 !important;
}

.nav-link.active {
  text-decoration: underline !important;
}
