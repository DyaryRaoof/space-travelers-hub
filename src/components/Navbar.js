import React from 'react';
import { NavLink } from 'react-router-dom';
import planetLogo from '../img/planet.png';
import './Navbar.css';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/profile',
      text: 'My profile',
    },
  ];

  return (
    <nav className="navbar navbar-expand-lg navbar-light panel-bg">
      <div className="d-flex container justify-content-between align-items-center">
        <div className="d-flex align-items-center">
          <img className="planet-logo" src={planetLogo} alt="Logo" />
          <span>Space Travelers&apos; Hub</span>
          <ul className="navbar-nav">
            {links.map((link) => (
              <li key={link.id} className="nav-item ps-3">
                <NavLink to={link.path} className="nav-link" exact="true">
                  {link.text}
                </NavLink>
              </li>
            ))}
          </ul>
        </div>
      </div>
    </nav>
  );
};
export default Navbar;
