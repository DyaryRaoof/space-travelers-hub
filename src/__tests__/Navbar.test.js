import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

test('renders correctly', () => {
  const rockets = renderer
    .create(<Router><Navbar /></Router>)
    .toJSON();
  expect(rockets).toMatchSnapshot();
});
