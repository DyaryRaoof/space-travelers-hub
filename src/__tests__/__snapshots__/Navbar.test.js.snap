// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light panel-bg border-bottom"
>
  <div
    className="d-flex container justify-content-between align-items-center"
  >
    <div
      className="d-flex align-items-center"
    >
      <img
        alt="Logo"
        className="planet-logo"
        src="planet.png"
      />
      <span
        className="header-title"
      >
        Space Travelers' Hub
      </span>
    </div>
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item ps-3"
      >
        <a
          aria-current="page"
          className="nav-link active"
          exact="true"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="nav-item ps-3"
      >
        <a
          className="nav-link"
          exact="true"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="nav-item ps-3"
      >
        <a
          className="nav-link"
          exact="true"
          href="/profile"
          onClick={[Function]}
        >
          My profile
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
